src/swe/swe_ceed.c

```diff
@@ -263,6 +291,45 @@ PetscErrorCode CreateSWEFluxOperator(Ceed ceed, RDyMesh *mesh, int num_boundarie
   PetscFunctionReturn(0);
 }
 
+// Gets the field representing Dirіchlet boundary values for the given boundary.
+PetscErrorCode SWEFluxOperatorGetDirichletBoundaryValues(CeedOperator flux_op, RDyBoundary boundary, CeedOperatorField *boundary_values) {
+  PetscFunctionBeginUser;
+
+  // get the relevant boundary sub-operator
+  CeedOperator *sub_ops;
+  CeedCompositeOperatorGetSubList(flux_op, &sub_ops);
+  CeedOperator boundary_flux_op = sub_ops[1 + boundary.index];
+
+  // fetch the field
+  CeedOperatorGetFieldByName(boundary_flux_op, "q_dirichlet", boundary_values);
+
+  PetscFunctionReturn(0);
+}
+
+PetscErrorCode SWEFluxOperatorSetDirichletBoundaryValues(CeedOperator flux_op, RDyBoundary boundary, PetscReal *boundary_values) {
```

Comment:
Should we also send `num_comp` and `num_edges` as input arguments so we know the size of `*boundary_values` AND an `assert` to check the size of  `*boundary_values`?

Reply:
Sure. This is a matter of taste, but as you point out it may help us find errors and inconsistencies.

Reply:
Something similar to https://github.com/RDycore/RDycore/pull/94/files#diff-2380356150436250abd03a1522da01310c2e79d6104107b17cb4667e1d723fe4R41

Reply:
```
PetscErrorCode SWEFluxOperatorSetDirichletBoundaryValues(CeedOperator flux_op, RDyBoundary boundary, PetscReal boundary_values[boundary.num_edges * ndof])
```

Reply:
I think it's okay to handle these checks in `RDySetDirichletBoundaryValues` for now, but if we start to see these sorts of bugs crop up in our internal code, we can always add these parameters.